src/util.cpp

```diff
@@ -639,6 +639,9 @@ void ArgsManager::ReadConfigFile(const std::string& confPath)
     }
     // If datadir is changed in .conf file:
     ClearDatadirCache();
+    if (!fs::is_directory(GetDataDir(false))) {
+        throw std::runtime_error(strprintf("specified data directory \"%s\" does not exist.", gArgs.GetArg("-datadir", "").c_str()));
```

Comment:
Any reason why you're once using `GetDataDir(false)` and once `gArgs.GetArg("-datadir", "")`? It's a bit confusing if they can ever differ, and if so, why they're used here.

Reply:
If the datadir doesn't exist I think GetDataDir returns an empty string so the argument needs to be accessed directly in that case